simtools/db_handler.py

```diff
@@ -219,33 +216,94 @@ def getModelParameters(
         _telModelNameValidated = names.validateTelescopeModelName(telescopeModelName)
 
         if cfg.get('useMongoDB'):
+
+            # Only MongoDB suppports tagged version
+            _modelVersion = self._convertVersionToTagged(
+                modelVersion,
+                DatabaseHandler.DB_CTA_SIMULATION_MODEL
+            )
+            _versionValidated = names.validateModelVersionName(_modelVersion)
+
             _pars = self._getModelParametersMongoDB(
                 DatabaseHandler.DB_CTA_SIMULATION_MODEL,
                 _siteValidated,
                 _telModelNameValidated,
                 _versionValidated,
-                runLocation,
                 onlyApplicable
             )
             return _pars
         else:
+            _versionValidated = names.validateModelVersionName(modelVersion)
+
             return self._getModelParametersYaml(
                 _siteValidated,
                 _telModelNameValidated,
                 _versionValidated,
                 onlyApplicable
             )
 
+    def exportModelFiles(self, parameters, dest):
+
+        def isfloat(value):
+            try:
+                float(value)
+                return True
+            except ValueError:
+                return False
+
+        if cfg.get('useMongoDB'):
+            self._logger.debug('Exporting model files from MongoDB')
+            for par, info in parameters.items():
+                if not info['File']:
+                    continue
+                file = self._getFileMongoDB(
+                    DatabaseHandler.DB_CTA_SIMULATION_MODEL,
+                    info['Value']
+                )
+                self._writeFileFromMongoToDisk(DatabaseHandler.DB_CTA_SIMULATION_MODEL, dest, file)
+        else:
+            self._logger.debug('Exporting model files from local model file directories')
+            for par, value in parameters.items():
+
+                shallNotSearch = (
+                    not isinstance(value, str)
+                    or isfloat(value)
+                    or '.' not in value
+                )
+
+                if shallNotSearch:
+                    continue
+                self._writeModelFileYaml(value, dest, noFileOk=True)
+
+    def _writeModelFileYaml(self, fileName, destDir, noFileOk=False):
+        '''
+        Find the fileName in the model files location and write a copy
+        at the destDir directory.
+
+        Parameters
+        ----------
+        fileName: str or Path
+            File name to be found and copied.
+        destDir: str or Path
+            Path of the directory where the file will be written.
+        '''
+
+        destFile = Path(destDir).joinpath(fileName)
+        try:
+            file = cfg.findFile(fileName, cfg.get('modelFilesLocations'))
+        except FileNotFoundError:
+            if noFileOk:
+                self._logger.debug('File not found but noFileOk')
```

Comment:
Add the name of the file to the message for easier debugging?

Reply:
Done